include/geos/geom/FixedSizeCoordinateSequence.h

```diff
@@ -105,7 +105,9 @@ namespace geom {
         }
 
         void setPoints(const std::vector<Coordinate> & v) final override {
-            std::copy(v.begin(), v.end(), m_data.begin());
+            if (N > 0 && v.size() <= N) {
```

Comment:
I would probably opt for a crash (status quo) or exception over a silent failure here.

Reply:
Good idea.  Is there an existing pattern for an exception?

Reply:
The existing pattern (in GEOS, as well as the STL) is to crash if the caller tries to access an index that does not exist, which essentially is what is happening here. You could throw a `GEOSException` if `v.size() != N`. An `assert` would be the middle ground.

Reply:
How would a crash work?  Does the original code crash, or just overwrite mem?  

In any case, need an explicit check for the new GCC.  So need to do something on failure.  I like `assert` best, since it makes the contract explicit.